crates/red_knot_python_semantic/src/semantic_index/expression.rs

```diff
@@ -35,6 +35,13 @@ pub(crate) struct Expression<'db> {
     #[return_ref]
     pub(crate) node_ref: AstNodeRef<ast::Expr>,
 
+    /// Whether or not this expression should be inferred as a type expression or
+    /// a normal expression. For example, in `self.x: <annotation> = <value>`, the
+    /// `<annotation>` is inferred as a type expression, while `<value>` is inferred
+    /// as a normal expression.
+    #[id]
```

Comment:
Reading the docs on [`#[id]`](https://salsa-rs.netlify.app/overview.html?highlight=%23%5Bid%5D#id-fields), it's still not clear to me if this is correct or not. So please review this change of the `Expression` struct.